packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -6,10 +6,26 @@
  */
 import Cookies from 'js-cookie'
 
-export interface BaseStorage {
-    set(key: string, value: string, options?: unknown): void
-    get(key: string): string
-    delete(key: string): void
+export type StorageTypes = 'cookie' | 'local'
+
+export interface BaseStorageOptions {
+    keyPrefix?: string
+    keyPrefixSeperator?: string
+}
+
+export abstract class BaseStorage {
```

Comment:
An abstract class with implementation, fancy typescript stuff huh.

Reply:
Ha.. had to dust of an old bank in my memory to remember that stuff from java